src/widgets/checkfields/checkphone.cpp

```diff
@@ -1,21 +1,48 @@
 #include "checkqlineedit.h"
 #include "checkphone.h"
-
-CheckPhone::CheckPhone(QWidget *w, QPushButton* btn) : CheckQLineEdit(w, btn) {
-
+#include <QDebug>
+CheckPhone::CheckPhone(QWidget *w, QPushButton* btn) : CheckUntilField(w, btn) {
+    _country = "";
 }
 
+bool CheckPhone::check(QString text)
+{
+    bool isChecked = true;
 
-bool CheckPhone::check(QString text) {
     if ( text.length() == 0 ) {
-        return false;
+        isChecked = false;
+    }
+
+    if (this->getCountry() == "FRANCE") {
+        text = text.replace(" ", "");
+        text = text.replace(".", "");
+        text = text.replace("+33","0");
+        text = text.replace("(","");
+        text = text.replace(")","");
+        if (text[0]=='0' && text[1]=='0' && text[2]=='3' && text[3]=='3') {
+            QString buff = text;
+            text = "0";
+            for (int i=0; i < 9; i++) {
+                text += ""+buff[i+4];
+            }
```

Comment:
Plutôt que faire un `for`, ça serait plus joli (et optimisé, mais bon, ça…) d'utiliser substring, non ? 
``` C++
QString myString("This is a string");
QStringRef subString(myString, 5, 2); // subString contains "is"
```
http://stackoverflow.com/questions/7584699/qt-get-part-of-qstring